spring-integration-core/src/main/java/org/springframework/integration/config/annotation/MessagingAnnotationPostProcessor.java

```diff
@@ -272,4 +277,30 @@ protected String generateBeanName(String originalBeanName, Method method,
 		return this.postProcessors;
 	}
 
+	@Override
+	public void destroy() throws Exception {
+		this.beanFactory.getBean(Disposables.class).destroy();
```

Comment:
This is not god to call `getBean()` from the `destroy()`.
See some recent cause here: https://stackoverflow.com/questions/52785517/beancreationnotallowedexception-from-predestroy-annotation-from-prometheuspushg

Reply:
Right - I have changed the registration so the container will do the destruction.